editor/effects.js

```diff
@@ -20,9 +20,30 @@ import {
 	getCurrentPostType,
 	getBlocks,
 	getPostEdits,
+	getDocumentTitle,
 } from './selectors';
 
+let originalDocumentTitle; // @todo Obtain document.title template from PHP export?
+function populateDocumentTitle( title ) {
+	if ( ! originalDocumentTitle ) {
+		originalDocumentTitle = document.title;
+	}
+	document.title = title + ' | ' + originalDocumentTitle;
+}
+
 export default {
+	RESET_POST( action, store ) {
+		setTimeout( () => { // Next-tick to ensure action has been applied to state in store .
```

Comment:
@aduth check out the latest changes. It feels much better. I looked at the `DocumentHead` component in Calypso and used it to inspire a `DocumentTitle` component.